src/satellitePropagator/physics/AccelerationAccumulator.h

```diff
@@ -245,83 +245,86 @@ void AccelerationAccumulator<Container>::applyComponents() const
-                if (config[C22]) {
-                    new_acc_component = C22Component::apply(d, c_term, s_term);
+            // Managed particles compensate for all pertubations and stay on the keplerian orbit. We simulate this by not applying any pertubations on them.
+            if(d.getActivityState() != Container::Particle_t::ActivityState::evasivePreserving) {
+                if (config[J2]) {
+                    new_acc_component = J2Component::apply(d);
                     if (write) {
                         file_output->writeAcc_value(new_acc_component);
                     }
                     new_acc_total[0] += new_acc_component[0];
                     new_acc_total[1] += new_acc_component[1];
                     new_acc_total[2] += new_acc_component[2];
                 }
-                if (config[S22]) {
-                    new_acc_component = S22Component::apply(d, c_term, s_term);
+                // if we want to calculate both C22 and S22 we can share many of the
+                // calculation steps
+                if (config[C22] && config[S22]) {
+                    new_acc_component = C22S22Component::apply(d, c_term, s_term);
                     if (write) {
                         file_output->writeAcc_value(new_acc_component);
+                        file_output->writeAcc_value(new_acc_component);
                     }
                     new_acc_total[0] += new_acc_component[0];
                     new_acc_total[1] += new_acc_component[1];
                     new_acc_total[2] += new_acc_component[2];
+                } // if only one of the two should be calculated we only calculate the
+                // needed one
+                else {
+                    if (config[C22]) {
+                        new_acc_component = C22Component::apply(d, c_term, s_term);
+                        if (write) {
+                            file_output->writeAcc_value(new_acc_component);
+                        }
+                        new_acc_total[0] += new_acc_component[0];
+                        new_acc_total[1] += new_acc_component[1];
+                        new_acc_total[2] += new_acc_component[2];
+                    }
+                    if (config[S22]) {
+                        new_acc_component = S22Component::apply(d, c_term, s_term);
+                        if (write) {
+                            file_output->writeAcc_value(new_acc_component);
+                        }
+                        new_acc_total[0] += new_acc_component[0];
+                        new_acc_total[1] += new_acc_component[1];
+                        new_acc_total[2] += new_acc_component[2];
+                    }
                 }
-            }
-            if (config[SOL]) {
-                new_acc_component = SolComponent::apply(d, d_srp, sun_params);
-                if (write) {
-                    file_output->writeAcc_value(new_acc_component);
+                if (config[SOL]) {
```

Comment:
The changelog seems to have gone a little haywire in this section, I'll nothing actually changed :D

Reply:
If you [hide changes to whitespaces](https://github.blog/2018-05-01-ignore-white-space-in-code-review/) you see that there didn't actually happen anything ;)